src/components/Message/hooks/useReactionHandler.ts

```diff
@@ -31,17 +31,28 @@ export const useReactionHandler = <
       reaction: ReactionResponse<StreamChatGenerics>,
       message: StreamMessage<StreamChatGenerics>,
     ): StreamMessage<StreamChatGenerics> => {
-      const newReactionCounts = message?.reaction_counts || {};
+      const newReactionGroups = message?.reaction_groups || {};
       const reactionType = reaction.type;
-      const hasReaction = !!newReactionCounts[reactionType];
+      const hasReaction = !!newReactionGroups[reactionType];
 
       if (add) {
-        newReactionCounts[reactionType] = hasReaction ? newReactionCounts[reactionType] + 1 : 1;
+        const timestamp = new Date().toISOString();
+        newReactionGroups[reactionType] = hasReaction
+          ? { ...newReactionGroups[reactionType], count: newReactionGroups[reactionType].count + 1 }
+          : {
+              count: 1,
+              first_reaction_at: timestamp,
+              last_reaction_at: timestamp,
+              sum_scores: 1,
+            };
```

Comment:
Optimistic updates for reaction groups. Not technically correct (because score defaults to 1, which might not be the case), but no more incorrect than an existing implementation ;)